Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/SecurityController.cs

```diff
@@ -1023,6 +1027,283 @@ public HttpResponseMessage GetLastModifiedSettings()
             }
         }
 
+        /// GET: api/Security/GetApiTokenSettings
+        /// <summary>Gets settings for API tokens.</summary>
+        /// <returns>API token settings.</returns>
+        [HttpGet]
+        [AdvancedPermission(MenuName = Components.Constants.MenuName, Permission = Components.Constants.ManageApiTokens)]
+        public HttpResponseMessage GetApiTokenSettings()
+        {
+            try
+            {
+                var response = new
+                {
+                    Success = true,
+                    Results = new
+                    {
+                        ApiTokenSettings = ApiTokenSettings.GetSettings(this.PortalId),
+                    },
+                };
+
+                return this.Request.CreateResponse(HttpStatusCode.OK, response);
+            }
+            catch (Exception exc)
+            {
+                Logger.Error(exc);
+                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, exc);
+            }
+        }
+
+        /// POST: api/Security/UpdateApiTokenSettings
+        /// <summary>Updates API Token settings.</summary>
+        /// <param name="request">The update request.</param>
+        /// <returns>A response indicating success.</returns>
+        [HttpPost]
+        [RequireAdmin]
+        [ValidateAntiForgeryToken]
+        public HttpResponseMessage UpdateApiTokenSettings(UpdateApiTokenSettingsRequest request)
+        {
+            try
+            {
+                var settings = ApiTokenSettings.GetSettings(this.PortalId);
+                if (this.UserInfo.IsSuperUser)
+                {
+                    settings.MaximumTimespan = request.MaximumTimespan;
+                    settings.MaximumTimespanMeasure = request.MaximumTimespanMeasure;
+                }
+
+                settings.AllowApiTokens = request.AllowApiTokens;
+                settings.SaveSettings(this.PortalId);
+                DataCache.ClearCache();

```

Comment:
Do we need to clear the whole cache whenever the API token settings are adjusted?